gravitee-am-service/src/main/java/io/gravitee/am/service/model/PatchDomain.java

```diff
@@ -220,8 +221,14 @@ public Domain patch(Domain _toPatch) {
             }
         }
 
-        if (this.passwordSettings != null) {
-            this.passwordSettings.ifPresent(ps -> toPatch.setPasswordSettings(ps.patch(toPatch.getPasswordSettings())));
+        if (passwordSettings != null && passwordSettings.isPresent()) {
+            var ps = passwordSettings.get();
+            if (ps.getInherited() != null && ps.getInherited().isPresent() && Boolean.TRUE.equals(ps.getInherited().get())) {
+                toPatch.setPasswordSettings(null);
+                setPasswordSettings(Optional.empty());
+            } else {
+                toPatch.setPasswordSettings(ps.patch(toPatch.getPasswordSettings()));
+            }
```

Comment:
**todo**: Not sure about this way to manage the reset. You are considering the presence of the inherited boolean but everywhere else into the code for patch/update action we consider the Optional.isEmpty() as the condition to reset the value to null. 

Basically, in the request payload:
* if the field is missing, the Java bean will have a null for this attribute,  in that case we do not change the value
* if the field is set to null, the Java bean will have a Optional.empty() for this attribute,  in that case we set the value to null
* if the field is not null, the Java bean will have a Optional.from() for this attribute,  in that case we change the value with the content of the Optional.

The change should be something like this IMO to be consistent

```suggestion
        if (passwordSettings != null) {
            if (passwordSettings.isEmpty()) {
                toPatch.setPasswordSettings(null);
            } else {
                toPatch.setPasswordSettings(this.passwordSettings.get().patch(toPatch.getPasswordSettings()));
            }
```
wdyt ?

Reply:
![image](https://github.com/gravitee-io/gravitee-access-management/assets/152597515/8bbf8126-a2e3-4990-a2c7-b2fe3eedbdd5)
![image](https://github.com/gravitee-io/gravitee-access-management/assets/152597515/043cbc54-f02c-4756-8488-04f3e6b65a14)

Reply:
ok, as discussed we can rely on null.